### ISSUE TITLE
ure: Incorrect match offsets with non-ASCII strings

### ISSUE THREAD

**Issue by jepler** [micropython/micropython#9202]
Code:
```python
import re

text1 = "MicroPython loves regular expressions"
text2 = "MicroPython \u2764 regular expressions"

regex1 = re.compile("re")

for text in (text1, text2):
    match = regex1.search(text)
    print(text)
    print("Match:", text[match.start():match.end()])
    print()
```

Result on standard Python:
```
$ python3.9 kevin.py 
MicroPython loves regular expressions
Match: re

MicroPython ❤ regular expressions
Match: re
```

Result on MicroPython v1.19.1-358-g0b26efe73-dirty on 2022-09-03; linux [GCC 10.2.1] version:
```
./build-coverage/micropython  code.py 
MicroPython loves regular expressions
Match: re

MicroPython ❤ regular expressions
Match: gu
```

Originally reported at https://github.com/adafruit/circuitpython/issues/6860

My _guess_ (not really an analysis) is that re is failing to convert byte offsets to code-point offsets when it is searching a utf-8 string instead of a bytes object.

**Comment by dpgeorge** [comment]
Fixed by e90b85cc98a24003f2d673bab2c255ab3dce66e7

**Comment by jepler** [comment]
Thanks!
